packages/app/src/Element/Thread.tsx

```diff
@@ -295,6 +292,32 @@ export default function Thread(props: ThreadProps) {
     return chains;
   }, [notes]);
 
+  const root = useMemo(() => {
+    const currentNote = parsedNotes.find(a => a.Id === urlNoteHex);
+
+    // current note is the root if the current note ID is a key in the chains Map
+    if (chains.has(urlNoteHex)) {
+      return currentNote;
+    }
+
+    let rootEventId = currentNote?.Thread?.Root?.Event;
+
+    // sometimes the root event ID is missing, and we can only take the happy path if the root event ID exists
+    if (rootEventId) {
+      return parsedNotes.find(a => a.Id === rootEventId);
+    }
+
+    // worst case we need to search for the root ID in the chains Map
+    for (const [key, value] of chains.entries()) {
+      if (value.find(a => a.Id === urlNoteHex)) {
+        rootEventId = key;
+        break;
+      }
+    }
+
+    return parsedNotes.find(a => a.Id === rootEventId);
```

Comment:
Actually, this logic isn't right. It just worked on the examples. Working on fixing it now.